lms/static/js/instructor_dashboard/data_download_2.js

```diff
@@ -0,0 +1,402 @@
+
+    PendingInstructorTasks = function() {
+        return window.InstructorDashboard.util.PendingInstructorTasks;
+    };
+
+    ReportDownloads = function() {
+        return window.InstructorDashboard.util.ReportDownloads;
+    };
+
+    DataDownloadV2 = (function() {
+        function InstructorDashboardDataDownload($section) {
+            var dataDownloadObj = this;
+            this.$section = $section;
+            this.$section.data('wrapper', this);
+            this.$list_problem_responses_csv_input = this.$section.find("input[name='problem-location']");
+            this.$download_display_text = $('.data-display-text');
+            this.$download_request_response_error = $('.request-response-error');
+            this.$download_display_table = $('.profile-data-display-table');
+            this.$reports_request_response = $('.request-response');
+            this.$reports_request_response_error = $('.request-response-error');
+            this.report_downloads = new (ReportDownloads())(this.$section);
+            this.instructor_tasks = new (PendingInstructorTasks())(this.$section);
+            this.$download_report = $('.download-report');
+            this.$gradeReportDownload = $('.grade-report-download');
+            this.$report_type_selector = $('.report-type');
+            this.$selection_informations = $('.selectionInfo');
+            this.$certificate_display_table = $('.certificate-data-display-table');
+            this.$downloadProblemReport = $('#download-problem-report');
+            this.$tabSwitch = $('.data-download-nav .btn-link');
+            this.$selectedSection = $('#' + this.$tabSwitch.first().attr('data-section'));
+            this.$learnerStatus = $('.learner-status');
+
+            this.clear_display = function() {
+                this.$download_display_text.empty();
+                this.$download_display_table.empty();
+                this.$download_request_response_error.empty();
+                this.$reports_request_response.empty();
+                this.$reports_request_response_error.empty();
+                this.$certificate_display_table.empty();
+                $('.msg-confirm').css({
+                    display: 'none'
+                });
+                return $('.msg-error').css({
+                    display: 'none'
+                });
+            };
+
+            this.clear_display();
+
+            this.$tabSwitch.click(function(event) {
+                var selectedSection = '#' + $(this).attr('data-section');
+                event.preventDefault();
+                $('.data-download-nav .btn-link').removeClass('active-section');
+                $('section.tab-data').hide();
+                $(selectedSection).show();
+                $(this).addClass('active-section');
+
+                $(this).find('select').trigger('change');
+                dataDownloadObj.$selectedSection = $(selectedSection);
+
+                dataDownloadObj.clear_display();
+            });
```

Comment:
We are registering this event(and some others below). It would be better to have a method like setupHandlers that is registering handlers against all the tabs and stuff.